src/core/sagas/products.ts

```diff
@@ -102,11 +111,20 @@ export namespace Tasks {
       yield effects.put(<any>flux.actions.receiveMoreProducts(e));
     }
   }
+
+  export function* fetchProductsWhenHydrated(flux: FluxCapacitor, action: Actions.fetchProductsWhenHydrated) {
+    if (Selectors.realTimeBiasesHydrated(flux.store.getState())) {
+      flux.store.dispatch(action.payload);
+    } else {
+      flux.once(Events.PERSONALIZATION_BIASING_REHYDRATED, () => flux.store.dispatch(action.payload));
```

Comment:
we don't need to access `flux` in order to do a dispatch here, you can use the redux effect `put()`

Reply:
Don't we still need flux for once?

Reply:
yea, but we should use the effects for `put()` as it makes it easier to test